### PR TITLE
rkt: add service

### PR DESCRIPTION
**Pull request by jgillich**

### CODE REVIEW

**Comment by zimbatm** [comment]
is the service running as root ?

**Comment by jgillich** [comment]
@zimbatm I think so, the socket is owned by root for authentication. Although we could add a `rkt` group similar to Docker.

**Comment by zimbatm** [comment]
I just had a look, Gentoo doesn't seem to be creating a special user[1] for it and imports the unit files from the project. It looks like the upstream repo also uses socket activation and has another rkt-gc service[2]

[1]: https://gitweb.gentoo.org/repo/gentoo.git/tree/app-emulation/rkt/rkt-0.8.0.ebuild
[2]: https://github.com/coreos/rkt/tree/master/dist/init/systemd

**Comment by jgillich** [comment]
gc sounds like a good idea, I've added it. Not sure about socket activation though, it's already failing for me with Docker (#11478). I think that can always be added at a later point.

**Comment by zimbatm** [comment]
:+1:

File: nixos/modules/virtualisation/rkt.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.rkt}/bin/rkt metadata-service";
+      };
+    };
+
+    systemd.services.rkt-gc = {
+      description = "rkt garbage collection";
+      startAt = "04:40";
```
Comment (jagajaga) [inline nixos/modules/virtualisation/rkt.nix]: I think it's better to make this option customizable..

Comment (jgillich) [inline nixos/modules/virtualisation/rkt.nix]: @jagajaga You mean gc in general or the execution time? At least currently, containers can't be reused, so you always want them cleaned up by gc. It's also a very fast process, which is why I just used the same time as `system.autoUpgrade` (which isn't customizable either).

Comment (jagajaga) [inline nixos/modules/virtualisation/rkt.nix]: No, I mean smth like https://github.com/NixOS/nixpkgs/blob/788800e437c4a0a25d95e217540bded68804b25e/nixos/modules/services/misc/nix-gc.nix#L23

**Comment by jgillich** [comment]
@jagajaga Ok now you can turn it off, configure the date and modify the options, which should make everyone happy. :)

**Pull request merged**